about_matching.exs

```diff
@@ -33,6 +33,20 @@ defmodule AboutMatching do
     assert_raise MatchError, fn -> [a , a] = [1 , __?] end
   end
 
+  think "Matched values overwrite variable values" do
+    a = 1
+    assert a == __?
+
+    [a, _] = [2, 3]
+    assert a == __?
+  end
+
+  think "Values can be pinned to prevent them from being overwritten" do
```

Comment:
I like that, so maybe "Values can be pinned to ensure the right-hand value meets your expectation"?